### PR TITLE
Check descriptor exists to handle cases when it is deleted.

### PR DESCRIPTION
**Pull request by wilr**
Handle cases better when `refreshDescriptor` is called and no descriptor is found.

### CODE REVIEW

File: code/jobs/DoormanQueuedJobTask.php [hunk 1]
```diff
@@ -185,6 +200,13 @@ public function canRunTask() {
...
 		$this->refreshDescriptor();
-		return $this->descriptor->JobStatus === QueuedJob::STATUS_CANCELLED;
+
+		$descriptor = $this->getDescriptor();
+
+		if ($descriptor) {
+			return $this->descriptor->JobStatus === QueuedJob::STATUS_CANCELLED;
+		}
+
+		return true;
```
Comment (emteknetnz) [inline code/jobs/DoormanQueuedJobTask.php]: This should be `return false`?

Missing descriptor doesn't really mean it was cancelled, just that it's missing?

File: code/jobs/DoormanQueuedJobTask.php [hunk 2]
```diff
@@ -168,14 +176,21 @@ public function shouldExpire($startedAt) {
...
-				QueuedJob::STATUS_WAIT
-			)
-		);
+
+		$descriptor = $this->getDescriptor();
+
+		if ($descriptor) {
+			return in_array(
+				$descriptor->JobStatus,
+				array(
```
Comment (emteknetnz) [inline code/jobs/DoormanQueuedJobTask.php]: should use short array syntax

**Review by emteknetnz — CHANGES_REQUESTED** [review]
Looks good, question around the 'default' value of isCancelled()

File: code/jobs/DoormanQueuedJobTask.php (hunk 1 above)
Comment (wilr) [reply code/jobs/DoormanQueuedJobTask.php]: Changed to false

File: code/jobs/DoormanQueuedJobTask.php (hunk 2 above)
Comment (wilr) [reply code/jobs/DoormanQueuedJobTask.php]: Updated to short syntax

**Comment by wilr** [comment]
@emteknetnz Updated based on feedback

**Comment by emteknetnz** [comment]
Great ta, I'll test this locally later today.

**Comment by emteknetnz** [comment]
Apologies, I didn't realise this was for Silverstripe 3 instead of Silverstripe 4

I probably gave bad advice re the short-array syntax, since 3.1 of this module requires framework ^3.1 it technically needs to support php 5.3 which does not support short array syntax =(

Technically we (Silverstripe Ltd who I work for) only have limited support for Silverstripe 3 issues meaning only critical and security related.  This pull-request isn't covered by that.  See https://www.silverstripe.org/software/roadmap/.

I guess I can still approve this one for you though based on code-review (it looks fine), though testing/merging will need to be a community thing

Sorry once again =/

**Comment by GuySartorelli** [comment]
The branch this is targeting is no longer supported.
If this affects Silverstripe CMS 5, please target the `5.1` branch.

**Comment by wilr** [comment]
Looks like it has been fixed independently. this->describe is checked before the call.

**Pull request closed without merging**